### DISCUSSION TITLE
Support EIP-6963 in Mobile Connector

### DISCUSSION THREAD

**Discussion by ablankstory** [family/connectkit#338] [Ideas]
Recently, you integrated support for EIP-6963 (https://github.com/family/connectkit/pull/312), however, it doesn't appear to work when connecting via a mobile wallet's dApp browser. It would be great if you could add support for this use case. We are prioritizing work for supporting eip-6963 in our wallet (Loopring Wallet), and would love for users to be able to easily find us at the top of the list when connecting via our dApp browser.

Example:

1. Use MetaMask or Ronin wallets and connect to web3modal.com using their dapp browser >> these wallets appear at the top of the list

2. Use MetaMask or Ronin wallets and connect to ConnectKit using their dapp browser >> they do not appear at the top of the list

Supporting this will make it much easier for users to find and connect to dapps when accessing via the dapp browser. It will remove all of the guess work when trying to find out which wallet to select if they don't see theirs on the list.

Thanks!

**Comment by lochie** [comment]
@ablankstory ConnectKit `1.6.0` and above should definitely be supporting EIP-6963 wallets and hoisting them to the top of the wallets list. 

Do you have a screen recording or any way to reproduce the issue you're seeing?
Also can you confirm that the wallets you are using are up to date and have implemented EIP-6963 support?

**Reply by ablankstory** [reply]
Thanks for the reply @lochie 

I could've sworn I tested this last week when I was looking at web3modal, RainbowKit, and ConnectKit. But when I tested today, ConnectKit appears to be working as expected for EIP-6963.

Although, it showed the wallet 4x at first. After refresh it's showing it 2x. Not sure if this is an issue with Ronin Wallet or not. Once our team (Loopring) implements support for 6963, I'll test with our wallet as well. 

Sorry for the confusion here!

![image](https://github.com/family/connectkit/assets/104391159/7ce91ced-22e7-4294-9252-b685e97cdef9)

![image](https://github.com/family/connectkit/assets/104391159/a2c97a28-a582-4dfc-9d6c-5be57931f3e3)


MetaMask dapp Browser only shows 1 instance of MM. so the issue may be isolated to Ronin Wallet - I'm not sure.

**Reply by lochie** [reply]
Oh yeah this definitely shouldn't happen.

A reason this might happen is Ronin might be injecting multiple versions of their provider. If that's the case then it's a due to them wanting to support both EIP-6963 and the previous [EIP-1193](https://eips.ethereum.org/EIPS/eip-1193) protocol.

We have done our best to provide a solution within ConnectKit to mitigate this doubling up of protocol support but we can only test wallets from an external viewpoint as we can't review their setup, so we're unsure what Ronin is exactly doing here to cause this.

**Reply by lochie** [reply]
Thanks for sharing the screenshots and re-reviewing these wallets @ablankstory 🙌
